tests/interpolation/built_in_resolvers/test_oc_select.py

```diff
@@ -0,0 +1,93 @@
+from omegaconf import OmegaConf
+
+
+def test_oc_select_abs() -> None:
+    cfg = OmegaConf.create(
+        {
+            "a0": "${k}",
+            "a1": "${oc.select:k}",
+            "a2": "${oc.select:k, zzz}",
+            "k": 10,
+        },
+    )
+    assert cfg.a0 == cfg.a1 == cfg.a2 == 10
+
+
+def test_oc_select_relative() -> None:
+    cfg = OmegaConf.create(
+        {
+            "a0": "${.k}",
+            "a1": "${oc.select:.k}",
+            "a2": "${oc.select:.k, zzz}",
+            "k": 10,
+        },
+    )
+    assert cfg.a0 == cfg.a1 == cfg.a2 == 10
+
+
+def test_oc_nested_select_abs() -> None:
+    cfg = OmegaConf.create(
+        {
+            "nested": {
+                "a0": "${k}",
+                "a1": "${oc.select:k}",
+                "a2": "${oc.select:k,zzz}",
+            },
+            "k": 10,
+        },
+    )
+
+    n = cfg.nested
+    assert n.a0 == n.a1 == n.a2 == 10
+
+
+def test_oc_nested_select_relative_same_level() -> None:
+    cfg = OmegaConf.create(
+        {
+            "nested": {
+                "a0": "${.k}",
+                "a1": "${oc.select:.k}",
+                "a2": "${oc.select:.k, zzz}",
+                "k": 20,
+            },
+        },
+    )
+
+    n = cfg.nested
+    assert n.a0 == n.a1 == n.a2 == 20
+
+
+def test_oc_nested_select_relative_level_up() -> None:
+    cfg = OmegaConf.create(
+        {
+            "nested": {
+                "a0": "${..k}",
+                "a1": "${oc.select:..k}",
+                "a2": "${oc.select:..k, zzz}",
+                "k": 20,
+            },
+            "k": 10,
+        },
+    )
+
+    n = cfg.nested
+    assert n.a0 == n.a1 == n.a2 == 10
+
+
+def test_oc_select_using_default() -> None:
+    cfg = OmegaConf.create(
+        {
+            "a0": "${oc.select:zz, 10}",
+            "a1": "${oc.select:.zz, 11}",
+            "a2": "${oc.select:zz, null}",
+            "a3": "${oc.select:zz, ${value}}",
+            "a4": "${oc.select:missing, 10}",
+            "value": 20,
+            "missing": "???",
```

Comment:
As mentioned in the doc comment, I think the missing case should also be tested without a default value.

Can you also add a test when selecting a node whose value is `None` to make sure it doesn't use the default value? (it's working fine, but would be good to have a test for it). What I mean is:
```python
cfg = OmegaConf.create({"x": None, "y": "${oc.select:x, 3}"})
assert cfg.y is None
```

Reply:
added.